mllib/src/test/scala/org/apache/spark/ml/classification/DecisionTreeClassifierSuite.scala

```diff
@@ -334,6 +334,51 @@ class DecisionTreeClassifierSuite extends SparkFunSuite with MLlibTestSparkConte
     assert(importances.toArray.forall(_ >= 0.0))
   }
 
+  test("should support all NumericType labels") {
```

Comment:
It's not a major issue, but there is a large amount of code duplication in all the tests. I know you've already tried to pare it down, but if it's possible to further reduce it by extracting out some methods, that would be cleaner.